src/foam/nanos/so/SystemOutageTask.js

```diff
@@ -19,8 +19,31 @@ foam.INTERFACE({
   package: 'foam.nanos.so',
   name: 'SystemOutageTask',
 
+  documentation: `
+    Inteface for SystemOutage Task.
+    Note that persist method needs to be called after change to task to persist the change.
+  `,
+
   javaImports: [
-    'foam.core.X'
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.logger.Loggers'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'id',
+      order: -20,
+      visibility: 'RO'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.so.SystemOutage',
```

Comment:
Thinking, how about another interface  SystemOutageRuleTask which is rule based. 
and add cleanup, save and such to it. 
then we have a simple task for messages and such, and another that is rule based.

Reply:
discussed in a chat and decided to keep it as it is as other tasks might need these features